Servers.yaml

```diff
@@ -0,0 +1,206 @@
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess
+
+  # IAM Instance Profile
+  EC2InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles:
+        - !Ref EC2InstanceRole
+  # Launch Template for EC2 Instances
+  TwitterLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    Properties:
+      LaunchTemplateName: TwitterServiceTemplate
+      LaunchTemplateData:
+        InstanceType: t2.micro
+        ImageId: ami-0014a768bde80541f
+        SecurityGroupIds:
+          - !Ref WebServerSecurityGroup
+        BlockDeviceMappings:
+          - DeviceName: /dev/xvda
+            Ebs:
+              VolumeSize: 8
+              VolumeType: gp2
+              DeleteOnTermination: true
+        IamInstanceProfile:
+          Arn: !GetAtt EC2InstanceProfile.Arn
+        UserData: 
+          Fn::Base64: !Sub |
+            #!/bin/bash
+            # Update the system
+            yum update -y
+
+            # Install Apache web server
+            yum install -y httpd
+
+            # Start and enable Apache
+            systemctl start httpd
+            systemctl enable httpd
+
+            # Create a simple index.html
+            echo "<html>
+            <head>
+              <title>Twitter Service</title>
+            </head>
+            <body>
+              <h1>Welcome to the Twitter Service</h1>
+              <p>This is a placeholder page for the Twitter application.</p>
+            </body>
+            </html>" > /var/www/html/index.html
+
+            # Ensure proper permissions
+            chmod 644 /var/www/html/index.html
+
+        TagSpecifications:
+          - ResourceType: instance
+            Tags:
+              - Key: Name
+                Value: TwitterServiceInstance
+
+  # Auto Scaling Group
+  TwitterAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      AutoScalingGroupName: TwitterServiceASG
+      LaunchTemplate:
+        LaunchTemplateId: !Ref TwitterLaunchTemplate
+        Version: !GetAtt TwitterLaunchTemplate.LatestVersionNumber
+      MinSize: 2
+      MaxSize: 2
+      DesiredCapacity: 2
+      VPCZoneIdentifier: 
+        Fn::Split:
+            - ","
+            - !ImportValue "PrivateSubnets"
+      TargetGroupARNs:
+        - !Ref ALBTargetGroup
+      Tags:
+        - Key: Name
+          Value: TwitterServiceInstance
+          PropagateAtLaunch: true
+      HealthCheckType: EC2
+      HealthCheckGracePeriod: 300
+
+# Security Group for Load Balancer
+  LBSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Allow HTTP traffic to the Load Balancer
+      VpcId:  
+        Fn::ImportValue: "VpcId"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: !Ref SshCidrIp
```

Comment:
The Ingress should allow only for traffic from ALB.
I don't think cidrIP is valid here, rather you need to use `SourceSecurityGroup` attribute